Kernel/Locking/Mutex.h

```diff
@@ -26,6 +28,11 @@ class Mutex {
 
 public:
     using Mode = LockMode;
+    // FIXME: Allow some customization
+    //  Note: The internal Spinlock might have a different Rank than the Mutex
+    //  Note: Making this a template would currently mess with Threads and our implementation
+    //        As we currently include Thread.h and Thread uses Mutexes in its API
```

Comment:
1st note: Does this might refer to something that should be done in the future? Because currently the spinlocks can't have a dfiferent rank.
2nd note: Does this cause an include cycle or what do you mean?

Reply:
IIRC Mutecies are part of the out-of-line API of Thread so slapping a template on them would make that harder